crates/ra_hir_def/src/find_path.rs

```diff
@@ -385,14 +420,15 @@ mod tests {
 
     #[test]
     fn different_crate_renamed() {
+        // Even if a local path exists, if the item is defined externally, prefer an external path.
         let code = r#"
             //- /main.rs crate:main deps:std
             extern crate std as std_renamed;
             <|>
             //- /std.rs crate:std
             pub struct S;
         "#;
-        check_found_path(code, "std_renamed::S");
+        check_found_path(code, "std::S");
```

Comment:
I don't think this actually works? Maybe I misremember, but I think the `extern crate` declaration actually renames the dependency, so you can't use the original name. (Also I wonder what happens if the `extern crate` declaration is in a submodule :thinking: )

Reply:
If the `extern crate` is in a submodule, You can only access it from that submodule, unless you use `pub`:

```rust
mod a {
    pub extern crate alloc;
}

use alloc::vec::Vec; // failed to resolve: use of undeclared type or module `alloc`
use a::alloc::vec::Vec; // works fine
```

Reply:
I've tested the code in the test, and this also works:

```rust
mod a {
    pub extern crate std;
}

use std::collections::HashMap;
use a::std::collections::HashMap as MashHap;
```

@bjorn3's example doesn't work because `alloc` is not in the extern prelude (apparently?). I'll check if this takes the extern prelude into account correctly.

Reply:
I think it's find to punt on `extern crate` not working exactly :)

Reply:
Yeah. I'm not sure how to test this either since the extern prelude seems to be generated automatically by the test code.